WordPress/Classes/ViewRelated/NUX/Post Signup Interstitial/PostSignUpInterstitialCoordinator.swift

```diff
@@ -0,0 +1,63 @@
+import Foundation
+
+private struct Constants {
+    static let userDefaultsKeyFormat = "PostSignUpInterstitial.hasSeenBefore.%@"
+}
+
+class PostSignUpInterstitialCoordinator {
+    private let database: KeyValueDatabase
+    private let userId: NSNumber?
```

Comment:
Why is this an optional? Seems like the init method that takes the userid is only used for testing. 
Will we have a situation where we wont have a userid?

Reply:
Yes, if you sign in using a self hosted site there will not be a WP.com account for the user which results in the userID being nil.